### PR TITLE
Pretty much done. 

### PR DESCRIPTION
**Pull request by chanbrandon999**

### CODE REVIEW

File: TicTacToe/src/tictactoe/TicTacToe.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import java.util.Scanner;
 
 public class TicTacToe {
```
Comment (cgallinaro) [inline TicTacToe/src/tictactoe/TicTacToe.java]: Don't forget a class description at the top of the file and descriptions of your variables.

File: TicTacToe/src/tictactoe/TicTacToe.java [hunk 2]
```diff
@@ -28,27 +34,33 @@ public TicTacToe() {
...
      * @param player The player to check if they won
      * @return Boolean if player has won or not
      */
     public boolean winnerCheck(char player) {
 
-        for (int i = 0; i < 3; i++) {   //Tests for horizontal and vertical wins 
+        /**
+         * Tests for horizontal and vertical wins 
+         */
+        for (int i = 0; i < 3; i++) {
```
Comment (cgallinaro) [inline TicTacToe/src/tictactoe/TicTacToe.java]: Good job using a loop here

File: TicTacToe/test/tictactoe/TicTacToeTest.java [hunk 3]
```diff
@@ -85,47 +85,74 @@ public void testPlayMove() {
     }
```
Comment (cgallinaro) [inline TicTacToe/test/tictactoe/TicTacToeTest.java]: Your test for isWinner only covers one of the possible 8 cases (not to mention testing any cases where isWinner is false).

You test for isFull should check that the method returns false if the board is not full.

File: TicTacToe/test/tictactoe/TicTacToeTest.java [hunk 4]
```diff
@@ -85,47 +85,74 @@ public void testPlayMove() {
...
 
     @Test
+    public void testInputToAi() {
```
Comment (cgallinaro) [inline TicTacToe/test/tictactoe/TicTacToeTest.java]: Your AI tests should be in a TicTacToeAITest.java file.

File: TicTacToe/src/tictactoe/TicTacToeAI.java [hunk 5]
```diff
@@ -21,12 +20,22 @@ public TicTacToeAI() {
                 aiBoard[row][col] = ' ';          //sets the character of each spot to a blank space 
             }
         }
+
+    }
+
+    public void inputToAI(char player, int row, int col) {         //Assigns both the player and AI's moves to the AI board.
```
Comment (cgallinaro) [inline TicTacToe/src/tictactoe/TicTacToeAI.java]: I like all of the documentation in this file!

File: TicTacToe/src/tictactoe/TicTacToeAI.java [hunk 6]
```diff
@@ -44,159 +53,61 @@ public TicTacToeAI() {
...
     /**
+     * Places move at any available winning space. Attempt to block each win the
+     * player tries to make if it exists. Place random middle, corner or side
+     * piece in that priority.
      *
-     * @return Returns coordinates to either win, block opponent, or a corner
-     * piece.
+     * @return Coordinates in the form of an array for desired spot. piece.
      */
     public int[] aiInputHard() {
```
Comment (cgallinaro) [inline TicTacToe/src/tictactoe/TicTacToeAI.java]: The hard AI is excellent!

File: TicTacToe/src/tictactoe/TicTacToeAI.java [hunk 7]
```diff
@@ -6,12 +6,11 @@
```
Comment (cgallinaro) [inline TicTacToe/src/tictactoe/TicTacToeAI.java]: It would have been nice to see a TicTacToeAITest class which tested some of your custom AI logic.

**Comment by cgallinaro** [comment]
Mark:

Problem 1: 4.5/5
Problem 2: 5/5
Problem 3: 4.5/5
Problem 4: 11/10 (2 bonus marks for extra AI work, but I deducted 1 mark for missing test coverage)

Total: 25/25

File: TicTacToe/src/tictactoe/TicTacToeAI.java (hunk 7 above)
Comment (chanbrandon999) [inline TicTacToe/src/tictactoe/TicTacToeAI.java]: I'm pretty sure I did add some test cases for the AI, but it was in the same class as the other tests. It did test some of the AI logic.

Comment (cgallinaro) [inline TicTacToe/src/tictactoe/TicTacToeAI.java]: I did see that there was a tiny set of test cases for your AI in the TicTacToeTest class.  While I applaud any effort spent testing, I think you will agree that you did not provide test coverage for a good portion of your AI code (especially for the harder AI modes).  On top of this, the assignment stated that you should have provided a TicTacToeAITest class for these tests (which would have made them easier to find).

**Pull request merged**